File: 2015-determine-whether-matrix-can-be-obtained-by-rotation/2015-determine-whether-matrix-can-be-obtained-by-rotation.java
class Solution {
    private int[][] rotate(int[][] mat){
        int n = mat.length;
        int[][] res = new int[n][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                res[n - j - 1][i] = mat[i][j];
            }
        }
        return res;
    }
    private boolean isEqual(int[][] mat, int[][] target){
        int n = mat.length;
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                if(mat[i][j] != target[i][j]){
                    return false;
                }
            }
        }
        return true;
    }
    private void printMatrix(int[][] mat){
        for(int i = 0; i < mat.length; i++){
            System.out.println(Arrays.toString(mat[i]));
        }
    }
    public boolean findRotation(int[][] mat, int[][] target) {
        if(isEqual(mat, target)){
            return true;
        }
        int[][] t = rotate(mat);
        for(int i = 0; i < 3; i++){
            if(isEqual(t, target)){
                return true;
            }
            if(i != 2){
                t = rotate(t);
            }
        }
        return false;
    }
}

File: 2015-determine-whether-matrix-can-be-obtained-by-rotation/2015-determine-whether-matrix-can-be-obtained-by-rotation.py
class Solution:
    def rotate(self, mat: List[List[int]]) -> List[List[int]]:
        n = len(mat)
        res = []
        for i in range(n):
            res.append([0] * n)
        for i in range(n):
            for j in range(n):
                res[n - j - 1][i] = mat[i][j]
        return res

    def findRotation(self, mat: List[List[int]], target: List[List[int]]) -> bool:
        if mat == target:
            return True
        n = len(mat)
        t = self.rotate(mat)
        for i in range(3):
            if t == target:
                return True
            if i != 2:
                t = self.rotate(t)
        return False